tutorials/math/random_number_generation.rst

```diff
@@ -196,6 +326,41 @@ We can apply similar logic from arrays to dictionaries as well::
         # The same metal may be selected multiple times in succession.
         return random_metal
 
+ .. code-tab:: csharp
+
+    struct MetalInfo
+    {
+        public MetalInfo(int quantity, int price)
+        {
+            this.quantity = quantity;
+            this.price = price;
+        }
+        public int quantity;
+        public int price;
+    }
+
+    enum MetalType
+    {
+        copper, 
+        silver, 
+        gold,
+    }
```

Comment:
```suggestion
        Copper, 
        Silver, 
        Gold,
    }

```

Enum members are PascalCase in C#, and there should be an extra line after the closing brace in this case.

However, there is a bigger problem: This code is too different from the GDScript version. Either the C# code should be more like the GDScript code, or the GDScript code should be more like the C# code, or both. The GDScript code is not using enums, the C# code is. The C# code is using a `struct`, which is not possible in GDScript so this part should be changed to be more like the GDScript code.

Reply:
This is how I was told on discord to make the dictionary in C#. If you know how to make the dictionary in a way that's more similar to GDScript show me and I'll add it.